dynup:
  ros__parameters:
    # Engine
    engine_rate: 240

    # end pose
    arm_extended_length: 0.4
    foot_distance: 0.2
    hand_walkready_pitch: 40.0
    hand_walkready_height: -0.17
    trunk_height: 0.42
    trunk_pitch: 0.2
    trunk_x_final: 0.001

    arm_side_offset_back: 0.148

    # Back
    arms_angle_back: 120.36
    com_shift_1: 0.051
    com_shift_2: 0.0
    foot_angle: 51.76
    hands_behind_back_x: 0.162
    hands_behind_back_z: 0.183
    leg_min_length_back: 0.253
    time_foot_ground_back: 0.536
    time_full_squat_hands: 0.172
    time_full_squat_legs: 0.196
    time_legs_close: 0.068
    trunk_height_back: 0.2  # if you decrease this, the robot's ass collides its ankles
    trunk_overshoot_angle_back: 5.95
    wait_in_squat_back: 1.0

    # Front
    arm_side_offset_front: 0.13
    hands_pitch: -65.86
    leg_min_length_front: 0.244
    max_leg_angle: 71.71
    time_foot_close: 0.0
    time_foot_ground_front: 0.132
    time_hands_front: 0.3
    time_hands_rotate: 0.3
    time_hands_side: 0.3
    time_to_squat: 0.924
    time_torso_45: 0.462
    trunk_overshoot_angle_front: -5.0
    trunk_x_front: 0.091  # do not increase this unless you want to burn a fuse
    wait_in_squat_front: 1.2

    time_walkready: 2.0
    # Rise
    rise_time: 1.0

    # Descend
    descend_time: 0.25

    # Stabilier
    stabilizing: False
    minimal_displacement: False
    stable_threshold: 0.05
    stable_duration: 100
    stabilization_timeout: 2.0

    # Visualizer
    spline_smoothness: 100
    display_debug: True

dynup_pid_trunk_fused_roll:
  ros__parameters:
    p: 0.36
    i: 0.8674
    d: 0.03735
    i_clamp: 0
    i_clamp_min: 0.0
    i_clamp_max: 0.0
    antiwindup: True
    publish_state: False
dynup_pid_trunk_fused_pitch:
  ros__parameters:
    p: -0.42
    i: -5.31646
    d: -0.008295
    i_clamp: 0.0
    i_clamp_min: 0.0
    i_clamp_max: 0.0
    antiwindup: True
    publish_state: False
